.travis.yml

```diff
@@ -39,33 +39,37 @@ env:
 sudo: false
 
 script:
-  - set -e
-  - if [ "$TRAVIS_OSX_IMAGE" = "xcode8.3" ]; then brew reinstall --build-from-source ruby; fi
-  - echo "Configuring MacVim" && echo -en "travis_fold:start:config\\r"
+  - set -o errexit
+  - |
+    if [[ "${TRAVIS_OSX_IMAGE}" = "xcode8.3" ]]; then
+      brew reinstall --build-from-source ruby
+    fi
+  - echo -e "\\033[33;1mConfiguring MacVim\\033[0m" && echo -en "travis_fold:start:configure\\r\\033[0K"
   - NPROC=$(getconf _NPROCESSORS_ONLN)
-  - ./configure $CONFOPT --enable-fail-if-missing
+  - ./configure ${CONFOPT} --enable-fail-if-missing
   - cat src/auto/config.mk
   - cat src/auto/config.h
-  - grep -q -- "-DDYNAMIC_PERL_DLL=\\\\\"$vi_cv_dll_name_perl\\\\\"" src/auto/config.mk
-  - grep -q -- "-DDYNAMIC_PYTHON_DLL=\\\\\"$vi_cv_dll_name_python\\\\\"" src/auto/config.mk
-  - grep -q -- "-DDYNAMIC_PYTHON3_DLL=\\\\\"$vi_cv_dll_name_python3\\\\\"" src/auto/config.mk
-  - echo -en "travis_fold:end:config\\r"
-  - echo "Building MacVim" && echo -en "travis_fold:start:build\\r"
-  - make -j$NPROC
-  - echo -en "travis_fold:end:build\\r"
-  - echo "Testing MacVim" && echo -en "travis_fold:start:test\\r"
-  - rm -f result; $VIMCMD -g -f -c "redir>result" -c "lua print(\"Test\")" -c "redir END" -c q; cat result; echo; grep -q -w Test result
-  - rm -f result; $VIMCMD -g -f -c "redir>result" -c "perl VIM::Msg(\"Test\")" -c "redir END" -c q; cat result; echo; grep -q -w Test result
-  - rm -f result; $VIMCMD -g -f -c "redir>result" -c "py import sys; print(\"Test\")" -c "redir END" -c q; cat result; echo; grep -q -w Test result
-  - rm -f result; $VIMCMD -g -f -c "redir>result" -c "py3 import sys; print(\"Test\")" -c "redir END" -c q; cat result; echo; grep -q -w Test result
-  - rm -f result; $VIMCMD -g -f -c "redir>result" -c "ruby puts(\"Test\")" -c "redir END" -c q; cat result; echo; grep -q -w Test result
-  - rm -f result;
+  - grep -q -- "-DDYNAMIC_PERL_DLL=\\\\\"${vi_cv_dll_name_perl}\\\\\"" src/auto/config.mk
+  - grep -q -- "-DDYNAMIC_PYTHON_DLL=\\\\\"${vi_cv_dll_name_python}\\\\\"" src/auto/config.mk
+  - grep -q -- "-DDYNAMIC_PYTHON3_DLL=\\\\\"${vi_cv_dll_name_python3}\\\\\"" src/auto/config.mk
+  - echo -en "travis_fold:end:configure\\r\\033[0K"
+  - echo -e "\\033[33;1mBuilding MacVim\\033[0m" && echo -en "travis_fold:start:build\\r\\033[0K"
+  - make -j${NPROC}
+  - echo -en "travis_fold:end:build\\r\\033[0K"
+  - set +o errexit
+  - echo -e "\\033[33;1mTesting MacVim\\033[0m" && echo -en "travis_fold:start:test\\r\\033[0K"
+  - macvim_excmd() { ${VIMCMD} -u NONE -i NONE -g -f -X -V1 -es "$@" -c 'echo ""' -c quit 2>&1; }
```

Comment:
Why remove the `redit>result` and grep in the end?

Reply:
I think these one-line tests should check whether each of if_xxx commands work properly or not (minimal test), and output should be tested in `make test`.